### PR TITLE
APEXMALHAR-2374 Calling scan recursively if the current path is a dir…

### PR DESCRIPTION
**Pull request by francisf**
…ectory

@bhupeshchawda  please review this.

### CODE REVIEW

File: library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java [hunk 1]
```diff
@@ -1143,6 +1149,16 @@ protected void setPartitionCount(int partitionCount)
     {
       this.partitionCount = partitionCount;
     }
+
+    public boolean isRecursive()
+    {
+      return recursive;
+    }
+
+    public void setRecursive(boolean recursive)
```
Comment (bhupeshchawda) [inline library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java:1168]: Javadocs for accessors

File: library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java [hunk 2]
```diff
@@ -106,7 +106,7 @@ public void testSinglePartiton() throws Exception
         lines.add("f" + file + "l" + line);
       }
       allLines.addAll(lines);
-      FileUtils.write(new File(testMeta.dir, "file" + file), StringUtils.join(lines, '\n'));
+      FileUtils.write(new File(testMeta.dir + "/subdir", "file" + file), StringUtils.join(lines, '\n'));
```
Comment (bhupeshchawda) [inline library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java]: Please add more test cases considering multi level paths and files within each path.

**Review by bhupeshchawda — COMMENTED** [review]
Have added couple of comments. Rest looks good.

**Review by bhupeshchawda — APPROVED** [review]
LGTM

File: library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java [hunk 3]
```diff
@@ -100,13 +100,15 @@ public void testSinglePartiton() throws Exception
...
+    String subdir = "";
     for (int file = 0; file < 2; file++) {
+      subdir += String.format("/depth_%d", file);
       HashSet<String> lines = Sets.newHashSet();
       for (int line = 0; line < 2; line++) {
         lines.add("f" + file + "l" + line);
       }
       allLines.addAll(lines);
-      FileUtils.write(new File(testMeta.dir, "file" + file), StringUtils.join(lines, '\n'));
+      FileUtils.write(new File(testMeta.dir + subdir, "file" + file), StringUtils.join(lines, '\n'));
```
Comment (chaithu14) [inline library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java:122]: Can you add the unit test for recursive is false ?

Comment (francisf) [reply library/src/test/java/com/datatorrent/lib/io/fs/AbstractFileInputOperatorTest.java:122]: @chaithu14  added testSinglePartitonRecursive test case.

**Comment by chaithu14** [comment]
Looks fine to me.

**Pull request merged**